### PR TITLE
Use spf13/cobra for docker push and docker pull

### PR DESCRIPTION
**Pull request by vdemeester**
Moves image command pull and push to `api/client/image/pull.go` and `api/client/image/push.go` and use cobra  🐍.

/cc @dnephin @thaJeztah @LK4D4 @cpuguy83 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

File: api/client/pull.go [hunk 1]
```diff
@@ -1,74 +1,14 @@
...
-		// Check if tag is digest
-		return cli.trustedPull(ctx, repoInfo, registryRef, authConfig, requestPrivilege)
-	}
-
-	return cli.imagePullPrivileged(ctx, authConfig, distributionRef.String(), requestPrivilege, *allTags)
-}
-
-func (cli *DockerCli) imagePullPrivileged(ctx context.Context, authConfig types.AuthConfig, ref string, requestPrivilege types.RequestPrivilegeFunc, all bool) error {
+// ImagePullPrivileged pulls the image and displays it to the output
+func (cli *DockerCli) ImagePullPrivileged(ctx context.Context, authConfig types.AuthConfig, ref string, requestPrivilege types.RequestPrivilegeFunc, all bool) error {
```
Comment (dnephin) [inline api/client/pull.go]: This method (and the similar `ImagePushPrivileged()` are only used by one function `trustedPush/pull` , which is also used in only a single file (this one).

What do you think about either moving `trustedPush/Pull` into the new `api/client/image` package (which would require making the `target` struct exported), or alternatively moving `ImagePullPrivileged` to `api/client/trust.go` so we can remove this old module.

Comment (vdemeester) [inline api/client/pull.go]: @dnephin yeah I was wondering about that, same goes for `push.go`. I think I'll move it to `api/client/trust.go` for now 😇

File: api/client/image/pull.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	flags := cmd.Flags()
+
+	flags.BoolVarP(&opts.all, "all-tags", "a", false, "Download all tagged images in the repository")
+	client.AddTrustedFlags(flags, true)
+
+	return cmd
+}
+
+func runPull(dockerCli *client.DockerCli, opts pullOptions) error {
+
```
Comment (WeiZhang555) [inline api/client/image/pull.go]: unnecessary blank line? :smile:

Comment (vdemeester) [inline api/client/image/pull.go]: maybe :joy_cat:

File: cli/usage.go [hunk 3]
```diff
@@ -20,7 +20,6 @@ var DockerCommandUsage = []Command{
 	{"login", "Log in to a Docker registry"},
 	{"logout", "Log out from a Docker registry"},
 	{"ps", "List containers"},
-	{"pull", "Pull an image or a repository from a registry"},
 	{"push", "Push an image or a repository to a registry"},
```
Comment (dmcgowan) [inline cli/usage.go]: Remove "push" here as well

Comment (vdemeester) [inline cli/usage.go]: Oh damn, I messed up on a rebase :sweat:

**Comment by cpuguy83** [comment]
LGTM

**Comment by dmcgowan** [comment]
LGTM and needs rebase

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**